rescript/dereplicate.py

```diff
@@ -53,21 +57,25 @@ def dereplicate(sequences: DNAFASTAFormat,
             derep_taxa, seqs_out = _dereplicate_taxa(
                 taxa, sequences, clustered_seqs, uc, mode=mode)
 
-            if rank_handles != 'disable':
-                rank_handles = _rank_handles[rank_handles]
+            if backfill:
+                # Sort user ranks relative to allowed_ranks, this ensures
+                # that the taxonomic ranks supplied by the user are in order
+                sorted_rank_handles = [p for r, p in
+                                       _allowed_ranks.items()
+                                       if r in rank_handles]
                 derep_taxa.loc[:, 'Taxon'] = derep_taxa['Taxon'].apply(
-                    _backfill_taxonomy, args=([rank_handles]))
+                    _backfill_taxonomy, args=([sorted_rank_handles]))
 
     return seqs_out, derep_taxa
 
 
 def _backfill_taxonomy(taxon, rank_handles):
-    formatted_taxon = taxon.split(';')
+    formatted_taxon = [t.strip() for t in taxon.split(';')]
     tax_len = len(formatted_taxon)
     if tax_len >= len(rank_handles):
         return taxon
     else:
-        return ';'.join(formatted_taxon + rank_handles[tax_len:])
+        return '; '.join(formatted_taxon + rank_handles[tax_len:])
```

Comment:
Yeah, I was going back and forth on this myself. I can retain the `;` w/o the space.

Reply:
I left a note / question in the `test_dereplicate.py` file. I think we should add a user defined delimiter, like `;` or `; `. In case the user would like to keep the spacing between the ranks consistent when backfilling.

Reply:
>maybe set up a function to autodetect how taxonomy strings should be formatted?

I was just thinking that I can simply search for variations of the `;` delimiter with and without spaces, then use the most abundant one for backfilling.

> (on the other hand we could just have all rescript actions always remove any leading/trailing spaces in all taxonomy strings,

In the prior iteration of this code, I was stripping space off of each taxonomic rank, and re-joining. But I see your point about cryptic issues. And the work of updating all the other rescript actions...

Or am I making much ado about nothing here? I'll think on it...

Reply:
this might be a problem that we just can't fix (as issues like joining with other databases, etc, are something that users should exercise due diligence over, and not something that we can control).

I am happy to just stick with the status quo to strip from all taxonomic ranks, and we could open an issue to provide a uniform solution across RESCRIPt if you think that we should provide the option. 

Just to say, I think this is why in the original implementation of `merge` I did not expose the ranks themselves but the "flavor" of taxonomy that the user wanted to use. Then the formatting comes pre-baked. Obviously only a half-baked solution 😂 , hence #122 .

Reply:
Or better yet, perhaps we can simply point users to `edit-taxonomy` or wrap that action into this action prior to it returning output?

Reply:
yeah I like that approach (documenting that spaces are stripped, and pointing to `edit-taxonomy`). I would not wrap the action itself (as that would entail always passing over the entire series an additional time), but the internal function could be called to edit strings inline if internal spaces are detected, if you like the autodetect idea.

Reply:
@nbokulich For the moment, I decided to not worry about the inconstancy of spaces surrounding `;`. We can discuss further offline if interested.